gateway/sds_gateway/api_methods/management/commands/replace_index.py

```diff
@@ -0,0 +1,825 @@
+            self.stdout.write(
+                self.style.ERROR(
+                    f"Unexpected error getting count for '{index_name}': {e!s}"
+                )
+            )
+            return -1
+
+    def handle(self, *args, **options):
+        """Execute the command."""
+        response = input(
+            "WARNING: This command is potentially destructive to existing "
+            "Capture and OpenSearch index data and should be used with "
+            "extreme caution.\n"
+            "Are you sure you want to continue? (y/N): "
+        ).lower()
+        if response != "y":
+            self.stdout.write(self.style.WARNING("Command cancelled."))
+            return
+
+        client: OpenSearch = get_opensearch_client()
+        capture_type = options["capture_type"]
+        index_name = options["index_name"]
+
+        try:
+            # delete duplicate captures, including docs in the original index
+            self.delete_duplicate_captures(client, capture_type, index_name)
+        except Exception as e:  # noqa: BLE001
+            self.stdout.write(
+                self.style.ERROR(f"Error deleting duplicate captures: {e!s}")
+            )
+            self.stdout.write(self.style.WARNING("Aborting command..."))
+            return
+
+        # refresh the index
+        client.indices.refresh(index=index_name)
+
+        try:
+            # Use timezone-aware datetime
+            timestamp = datetime.now(UTC).strftime("%Y%m%d%H%M%S")
+            backup_index_name = f"{index_name}-backup-{timestamp}"
+
+            # get queryset of captures to reindex
+            captures = Capture.objects.filter(
+                capture_type=capture_type,
+                index_name=index_name,
+                is_deleted=False,
+            )
+
+            new_index_config = {
+                "mappings": get_mapping_by_capture_type(capture_type),
+                "settings": {
+                    "index": {
+                        "number_of_shards": 1,
+                        "number_of_replicas": 1,
+                    },
+                },
+            }
+
+            # Get original document count
+            original_count = self.get_doc_count(client, index_name)
+            if original_count < 0:
+                self.stdout.write(
+                    self.style.ERROR(f"Skipping {index_name} due to count error")
+                )
+                return
+
+            # create backup index with same mapping
+            try:
+                self.clone_index(client, index_name, backup_index_name)
+            except (RequestError, OpensearchConnectionError) as e:
+                self.stdout.write(
+                    self.style.ERROR(f"Failed to create backup index: {e!s}")
+                )
+                return
+
+            # delete original index and recreate it with new mapping
+            self.delete_index(client, index_name)
+            self.create_index(client, index_name, new_index_config)
+
```

Comment:
It'd be good to assert `self.get_doc_count(client, backup_index_name) == original_count` here (or a similar check) before we delete the original index.